src/reqs.c

```diff
@@ -61,6 +61,41 @@
 #ifdef UPSTREAM_SUPPORT
 #  define UPSTREAM_CONFIGURED() (config.upstream_list != NULL)
 #  define UPSTREAM_HOST(host) upstream_get(host, config.upstream_list)
+static const char base64[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+static void encode_base_64(char* src, char* dest, int max_len) {
+        int n, l, i;
+        l = strlen(src);
+        max_len = (max_len - 1) / 4;
+        for (i = 0; i < max_len; i++, src += 3, l -= 3) {
+                switch (l) {
+                        case 0:
+                                break;
+                        case 1:
+                                n = src[0] << 16;
+                        *dest++ = base64[(n >> 18) & 077];
```

Comment:
this is the base64 implementation taken from proxychains (of which i happen to be the maintainer), but it seems you made some white-space errors while copy/paste-ing it